### PR TITLE
[*] CORE: Fix unreachable return statements

### PR DESCRIPTION
**Pull request by gskema**
- `hasDelivery()` probably always returned false?;
- `application/json` - I assume this was the intended type?

### CODE REVIEW

**Comment by gRoussac** [comment]
Hi,

Nice one. I believe with the ; it always retuned $res ! Thanks for your PR.

Regards

File: classes/order/Order.php [hunk 1]
```diff
@@ -1967,19 +1967,19 @@ public function hasInvoice()
...
 	/**
-	 *
 	 * Has Delivery return true if this order has already a delivery slip
+	 *
 	 * @return bool
 	 */
 	public function hasDelivery()
 	{
-		if ($res = (int)Db::getInstance()->getValue('
+		$res = (int)Db::getInstance()->getValue('
```
Comment (jeromenadaud) [inline classes/order/Order.php]: Yes like  @jocel1 says it will be better a bool cast here

Comment (gskema) [inline classes/order/Order.php]: Ok, I'll update this.

**Comment by gRoussac** [comment]
Dear both,

https://github.com/PrestaShop/PrestaShop/pull/3986

Regards

**Pull request merged**